### PR TITLE
Has dedicated db

### PR DESCRIPTION
**Pull request by sedroche**
# Motivation

Ticket: https://issues.jboss.org/browse/RHMAP-15811
The current check for whether the database has been upgraded can be inaccurate.

# Solution
Add a new API that checks the database directly for the presence of the apps dedicated database.

### CODE REVIEW

File: lib/handlers/api.js [hunk 1]
```diff
@@ -36,6 +37,16 @@ router.post('/apps/:domain/:environment/:appname/migratedb', fhmbaasMiddleware.a
   return res.json(req.createDbResult);
 });
 
+router.get('/apps/:domain/:environment/:appname/dedicated/db', function(req, res, next) {
+  mongoUtils.hasDedicatedDb(req.params, function(err, hasDedicatedDb) {
```
Comment (fh-qube) [inline lib/handlers/api.js]: BLOCKER Add a semicolon at the end of this statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* BLOCKER 1 blocker

Watch the comments in this conversation to review them.

File: lib/util/mongo.js [hunk 2]
```diff
@@ -128,7 +129,34 @@ function getAdminDbUrl(mongodbUrl, adminAuth) {
...
+    if (err) {
+      return handleError(null, err, 'cannot open mongodb connection', cb);
+    }
+
+    db.admin().listDatabases(function(err, result) {
+      if (err) {
+        return handleError(null, err, 'can not authenticate admin user', cb);
+      }
+
+      const hasDedicatedDb = result.databases.some(db => db.name === params.domain + '-' + params.appname + '-' + params.environment);
```
Comment (pb82) [inline lib/util/mongo.js]: Shouldn't this be the guid of the app instead of `appname`?

Comment (pb82) [inline lib/util/mongo.js]: Nevermind, that's handled in the params.

File: lib/util/mongo.js [hunk 3]
```diff
@@ -128,7 +129,34 @@ function getAdminDbUrl(mongodbUrl, adminAuth) {
...
+ * Check whether an app has a dedicated database for an environment.
+ * @param {object} params.
+ * @param {string} params.domain - The domain name.
+ * @param {string} params.appname - The app id.
+ * @param {string} params.environment - The environment name.
+ */
+function hasDedicatedDb(params, cb) {
+  logger.trace(params, 'is dedicated database');
+
+  MongoClient.connect(fhmbaasMiddleware.config().mongoUrl, function(err, db) {
```
Comment (pb82) [inline lib/util/mongo.js]: Do we need to create a new connection here? Can't we reuse the mongoose connection from here https://github.com/feedhenry/fh-mbaas/blob/master/fh-mbaas.js#L257 ?

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* BLOCKER 1 blocker

Watch the comments in this conversation to review them.

File: lib/util/mongo.js [hunk 4]
```diff
@@ -128,7 +129,36 @@ function getAdminDbUrl(mongodbUrl, adminAuth) {
...
+    if (err) {
+      return handleError(null, err, 'cannot open mongodb connection', cb);
+    }
+
+    db.admin().listDatabases(function(err, result) {
+      if (err) {
+        return handleError(null, err, 'can not authenticate admin user', cb);
```
Comment (pb82) [inline lib/util/mongo.js]: `db.close()` needs to be called here too

**Comment by sedroche** [comment]
Retest this please

**Pull request merged**